source/dhtslib/sam.d

```diff
@@ -758,24 +766,27 @@ struct SamWriter{
                         0  use no extra threads
                         >1 add indicated number of threads (to a default of 1)
     */
-    this(T)(T fn,bam_hdr_t * header, int extra_threads = -1)
+    this(T)(T fn,bam_hdr_t * header, SAMWriterTypes t=SAMWriterTypes.BAM,int extra_threads = -1)
```

Comment:
I like enum and template, but shouldn't it instead operate based on file extension?
OTOH would need this if support writing to stream like stdout

Reply:
I used the enum for the case of stdout. We could potentially implement both? Use the enum for special files/streams and use the extension on normal files. Might be tricky to explain.

Reply:
please change template param name to f from fn

Reply:
Would be very easy to forget to pass the parameter, which defaults to bam, when trying to open sam or cram file.

EIther 1. do not give default value of BAM and leave no default (so its a compile error to not specify) or 2. create new enum value called Deduce or similar, and make that default

Reply:
adressed in 6cf7650

Reply:
resolved 
6cf7650